File: src/main/java/com/saomc/saoui/themes/elements/ElementParent.java
package com.saomc.saoui.themes.elements;

import com.saomc.saoui.themes.util.HudDrawContext;

/**
 * Part of saoui by Bluexin.
 *
 * @author Bluexin
 */
public interface ElementParent {
    double getX(HudDrawContext ctx);

    double getY(HudDrawContext ctx);

    double getZ(HudDrawContext ctx);
}

File: src/main/java/com/saomc/saoui/themes/elements/ElementParent.kt
package com.saomc.saoui.themes.elements

import com.saomc.saoui.api.themes.IHudDrawContext

/**
 * Part of saoui by Bluexin.

 * @author Bluexin
 */
interface ElementParent {
    fun getX(ctx: IHudDrawContext): Double

    fun getY(ctx: IHudDrawContext): Double

    fun getZ(ctx: IHudDrawContext): Double

    val name: String
}
